### ISSUE TITLE
Some types of conditional formatting rules lead to a corrupted Excel file

### ISSUE THREAD

**Issue by sandraros** [abap2xlsx/abap2xlsx#1265]
To reproduce, run this program:
```abap
REPORT z_issue_1265.
DATA: gc_save_file_name TYPE string VALUE 'issue_1265.xlsx'.
INCLUDE zdemo_excel_outputopt_incl.
START-OF-SELECTION.
  TRY.
      DATA lo_style_cond TYPE REF TO zcl_excel_style_cond.
      DATA(lo_excel) = NEW zcl_excel( ).
      DATA(lo_worksheet) = lo_excel->get_active_worksheet( ).

      lo_worksheet->set_cell( ip_columnrow = 'A1' ip_value = 'ABCD' ).

      DATA(lo_fill_solid_red) = lo_excel->add_new_style( ).
      lo_fill_solid_red->fill->filltype    = zcl_excel_style_fill=>c_fill_solid.
      lo_fill_solid_red->fill->bgcolor-rgb = zcl_excel_style_color=>c_red.

      " Add a conditional formatting rule
      lo_style_cond = lo_worksheet->add_new_style_cond( ).
      lo_style_cond->rule              = zcl_excel_style_cond=>c_rule_textfunction.
      lo_style_cond->mode_textfunction = VALUE #( textfunction = zcl_excel_style_cond=>c_textfunction_beginswith
                                                  text         = 'A'
                                                  cell_style   = lo_fill_solid_red->get_guid( ) ).
      lo_style_cond->priority          = 1.
      lo_style_cond->set_range( ip_start_column = 'A'
                                ip_start_row    = 1
                                ip_stop_column  = 'A'
                                ip_stop_row     = 1 ).

      lcl_output=>output( cl_excel = lo_excel ).
    CATCH cx_root INTO DATA(lx_root).
      MESSAGE lx_root TYPE 'I' DISPLAY LIKE 'E'.
  ENDTRY.
```

I noticed that the issue is solved by adding this block (a rule "cell is" i.e. a condition based on a basic mathematical operator) anywhere before the output:
```
      lo_style_cond = lo_worksheet->add_new_style_cond( ).
      lo_style_cond->rule        = zcl_excel_style_cond=>c_rule_cellis.
      lo_style_cond->mode_cellis = VALUE #( formula    = '"A"'
                                            operator   = zcl_excel_style_cond=>c_operator_equal
                                            cell_style = lo_fill_solid_red->get_guid( ) ).
      lo_style_cond->priority = 1.
      lo_style_cond->set_range( ip_start_column = 'A'
                                ip_start_row    = 1
                                ip_stop_column  = 'A'
                                ip_stop_row     = 1 ).
```

The bug seems to be in the method `create_xl_styles` which doesn't call the method `create_dxf_style` in case `lo_style_cond->rule` is any other value than these 4 class constants `zcl_excel_style_cond`: `c_rule_cellis`, `c_rule_expression`, `c_rule_top10` and `c_rule_above_average`. Concerned lines:
https://github.com/abap2xlsx/abap2xlsx/blob/b3201166fcfa14eb4bbc4f69007ef2e8cd660863/src/zcl_excel_writer_2007.clas.abap#L5347-L5408

Adding this block in these lines solves the issue (I will propose a fix):
```abap
          WHEN zcl_excel_style_cond=>c_rule_textfunction.
            me->create_dxf_style( EXPORTING
                                    iv_cell_style    = lo_style_cond->mode_textfunction-cell_style
                                    io_dxf_element   = lo_element
                                    io_ixml_document = lo_document
                                    it_cellxfs       = lt_cellxfs
                                    it_fonts         = lt_fonts
                                    it_fills         = lt_fills
                                  CHANGING
                                    cv_dfx_count     = lv_dfx_count ).
```

NB: that would solve one type of conditional rule, but there is a total of 10 `c_rule_*` constants, so maybe there are left 5 other types possibly buggy.

**Comment by sandraros** [comment]
I'm closing this issue because I have added the proposed fix to #1255.

**Comment by darnoc312** [comment]
Hello,

Only 3 of the left 5 other types are relevant.
- c_rule_none is none.
- c_rule_containstext seems to be an obsolte and unused first try of c_textfunction_containstext.
In case of texfunction the c_textfunction* constants are used as rule type in XML Output.
Please refer to class lcl_create_xl_sheet lines 930 to 938.

And the 3 remaining types (c_rule_databar, c_rule_iconset and c_rule_colorscale) are not concerning dxfid.
At least lcl_create_xl_sheet's method add_conditional_formatting does not use the dxfid attribute of XML Node "cfRule" in case of these types.

So in my opinion this seems to be sufficient.
